### DISCUSSION TITLE
[Feature] Estimate location from Google Takeout Location History

### DISCUSSION THREAD

**Discussion by andreagdipaolo** [immich-app/immich#3909] [Feature Request]
### The feature

Google allows you to export the location history of your account with [Google Takeout](https://takeout.google.com/).
Some media taken by your phone may not have a geolocation in their metadata, because of past choices that now you regret, or by mistake, or because you've taken it with a third party app that saves less metadata like a social network.
Also, some standalone cameras don't have GPS while you probably had the phone in your pocket while you were shooting.

Immich could guess the location of some photos by checking if there is a reasonably accurate position in the location history.
It could be triggered globally and/or by album, by period, by camera model, or even by selecting individual media.

Google Photos already does this.

I believe it's a better approach than [another feature request](https://github.com/immich-app/immich/discussions/1675) based on other photos' location.

There's already a couple standalone scripts [1](https://chuckleplant.github.io/2018/07/23/google-photos-geotag.html), [2](https://github.com/Symbianx/google-takeout-photo-location-fixer).
I don't know if it would be better to store the guessed location separately or just write it in the metadata.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
The functionality in terms of working out photos' locations is pretty interesting, but this seems pretty tightly coupled to having Google follow you everywhere, which I think is something most of us are trying to get away from.

**Reply by andreagdipaolo** [reply]
I believe that there are many Immich users that just don't want to pay for additional Google One space and still use their free Google account for things like maps, gmail, etc.

I'm one of them, but for example this feature would be useful to me for many photos taken some years ago without location enabled, and I don't remember why. If I see them in Google Photos, it's there, correctly estimated via the location history.
If I import them to Immich (the original jpegs from DCIM folder), that location data is lost.

Surely, many other users want to entirely delete Google from their life, but that's not everyone's will

**Reply by AngelaDMerkel** [reply]
It's possible to convert the takeout location data to GPX and then use `exiftool` or another utility to tag your images with some threshold for error. For example, any photos taken within 120 seconds of the last location recorded are tagged with that location. This is a good solution that doesn't require any effort on the developers part, and allows the user to derive location from other sources including a GPS receiver, for example.

**Comment by Llerr** [comment]
Or you can take the coordinates from the phone photo and add them to the camera photo.
